pkg/scheduler/framework/session_plugins.go

```diff
@@ -317,6 +321,47 @@ func (ssn *Session) IsTaskAllocationOnNodeOverCapacityFn(task *pod_info.PodInfo,
 	}
 }
 
+func (ssn *Session) SubsetNodesFn(podGroup *podgroup_info.PodGroupInfo, tasks []*pod_info.PodInfo, initNodeSet node_info.NodeSet) ([]node_info.NodeSet, error) {
+	nodeSets := []node_info.NodeSet{initNodeSet}
+	for _, subsetNodesFn := range ssn.SubsetNodesFns {
+		log.InfraLogger.V(7).Infof(
+			"Running plugin func <%v> on podGroup <%s/%s>", subsetNodesFn, podGroup.Namespace, podGroup.Namespace)
+		newNodeSets := []node_info.NodeSet{}
+		shouldSkipPlugin := false
+		for _, nodeSet := range nodeSets {
+			isRelevant, nodeSubsets, err := subsetNodesFn(podGroup, tasks, nodeSet)
```

Comment:
The flow here potentially allows a plugin to partition several nodesets and then skip on the next one - it is weird, but possible. Maybe we should introduce two extension points:
1 - one that will check if it is relevant
2- partitions the nodes (only if relevant)

There are cons in this method, because it is an option